src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,280 @@
+package com.minecolonies.coremod.inventory.api;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagInt;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.IWorldNameable;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+/**
+ * Abstract class wrapping around multiple IItemHandler.
+ */
+public abstract class AbstractCombinedItemHandler
+  implements IItemHandlerModifiable, INBTSerializable<NBTTagCompound>, IWorldNameableModifyable
+{
+
+    ///NBT Constants
+    private static final String NBT_KEY_HANDLERS           = "Handlers";
+    private static final String NBT_KEY_HANDLERS_INDEXLIST = "Index";
+    private static final String NBT_KEY_NAME               = "Name";
+
+    private final IItemHandlerModifiable[] handlers;
+    @NotNull
+    private final String                   defaultName;
+    @Nullable
+    private       String                   customName;
+
+    /**
+     * Method to create a new combined item handler.
+     *
+     * @param defaultName The default name of this Handler.
+     * @param handlers    The combining handlers.
+     */
+    protected AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final IItemHandlerModifiable... handlers)
+    {
+        this.handlers = handlers;
+        this.defaultName = defaultName;
+    }
+
+    private AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final String customName, @NotNull final IItemHandlerModifiable... handlers)
+    {
+        this.handlers = handlers;
+        this.customName = customName;
+        this.defaultName = defaultName;
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT()
+    {
+        final NBTTagCompound compound = new NBTTagCompound();
+
+        int index = 0;
+        final NBTTagList handlerList = new NBTTagList();
+        final NBTTagList indexList = new NBTTagList();
+        for (final IItemHandlerModifiable handlerModifiable : handlers)
+        {
+            if (handlerModifiable instanceof INBTSerializable)
+            {
+                final INBTSerializable serializable = (INBTSerializable) handlerModifiable;
+                handlerList.appendTag(serializable.serializeNBT());
+                indexList.appendTag(new NBTTagInt(index++));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS881)